osquery/tables/applications/linux/docker_deb_packages.cpp

```diff
@@ -0,0 +1,192 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <sys/socket.h>
+#include <sys/types.h>
+
+#include <boost/algorithm/string.hpp>
+
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <osquery/tables/applications/linux/namespace_ops.h>
+#include <osquery/tables/applications/posix/docker_api.h>
+#include <osquery/tables/system/linux/deb.h>
+
+#include <rapidjson/filereadstream.h>
+
+namespace osquery {
+namespace tables {
+
+static constexpr size_t BUF_SIZE{65536};
+
+void extractDebPackageInfo(const struct pkginfo* pkg,
+                           rapidjson::Writer<rapidjson::StringBuffer>& w) {
+  struct varbuf vb;
+  varbuf_init(&vb, 20);
+  w.StartObject();
+
+  // Iterate over the desired fieldinfos, calling their fwritefunctions
+  // to extract the package's information.
+  for (const struct fieldinfo& fip : fieldinfos) {
+    fip.wcall(&vb, pkg, &pkg->installed, fw_printheader, &fip);
+
+    std::string line = vb.string();
+    if (!line.empty()) {
```

Comment:
We will have to reset `varbuf` too before we continue, so leaving the logic to what it was before since it's cleaner to just skip the lines which are empty and resetting the `varbuf` after all the operations.